plugins/Core-Extensions/ProductLine/edu.toronto.cs.se.mmint.productline.design/src/edu/toronto/cs/se/mmint/productline/design/ToProductLine.java

```diff
@@ -250,62 +243,148 @@ private ContainerMapping createProductLineContainerMapping(ContainerMapping orig
 	    return containerMapping;
 	}
   
-  private NodeCreationDescription createProductLineNodeCreationTool(NodeCreationDescription originalCreateTool, Map<String, Object> map) {
+  private NodeCreationDescription createProductLineNodeCreationTool(NodeCreationDescription originalCreateTool, Map<String, DiagramElementMapping> map) {
 	    var createTool = this.out.dToolFactory.createNodeCreationDescription();
 	    createTool.setName("PL" + originalCreateTool.getName());
 	    createTool.setLabel("PL" + originalCreateTool.getLabel());
-	    String newString = originalCreateTool.getLabel().replace(" ", "");
-	    /*
-	     * QUESTION 4
-	     */
+	    
 	    for (var originalMapping : originalCreateTool.getNodeMappings()) {
-	    	/*
-	    	 * QUESTION 5
-	    	 */
 	    	NodeMapping newMapping = (NodeMapping) map.get(originalMapping.getName());
 	    	createTool.getNodeMappings().add(newMapping);
 	    }
-	    
-	    createTool.setIconPath("/edu.toronto.cs.se.modelepedia.statemachine.edit/icons/full/obj16/"+newString+".gif");
-//	    originalCreateTool.getInitialOperation().getFirstModelOperations();
 	    /*
-	     * QUESTION 7
+	     * QUESTION 1: How do I add these to the actual creation tool? Are these the right elements in the first
+	     * place?
```

Comment:
I'm sorry, hard to say without a computer.
When you click on them in the diagram spec, the properties view should show what type of variable they are (or opening the diagram spec as a text file, their xsi:type field will show it).

Reply:
<variable name="container"/>
<viewVariable name="containerView"/>

Yeah I already checked that. This is what it says.
In the drop down I have createNameVariable(), createContainerViewVariable(), createSelectContainerVariable(), createAcceleoVariable(), createElementVariable(), createElementSelectVariable(), createElementDropVariable(), createElementViewVariable(). There is no createVariable(). 

![image](https://github.com/user-attachments/assets/62add246-89f6-4c88-b3d5-88424deb131b)
This is what the properties view looks like too. Hence, the confusion.

Reply:
Also, when I try using createTool.setVariable()
The method setVariable(NodeCreationVariable) in the type NodeCreationDescription is not applicable for the arguments (NameVariable)

I get this but I can't seem to be able to create a NodeCreationVariable

Reply:
Yep, my best guess is that a NodeCreationVariable can be created from another factory.
What about the dToolFactory?